### PR TITLE
Send resolution to subscriber

### PR DESCRIPTION
**Pull request by lukasz-szyszkowski**

### CODE REVIEW

File: Examples/SubscriberExample/Sources/Screens/Map/MapViewController.swift [hunk 1]
```diff
@@ -224,4 +224,8 @@ extension MapViewController: SubscriberDelegate {
         assetStatusLabel.textColor = statusDescAndColor.color
         assetStatusLabel.text = statusDescAndColor.desc
     }
+    
+    func subscriber(sender: Subscriber, didUpdateEnhancedLocation resolution: Resolution) {
+        // TODO
+    }
```
Comment (KacperKluka) [inline Examples/SubscriberExample/Sources/Screens/Map/MapViewController.swift:230]: Why is it added here? :thinking: Also the second param name probably shouldn't be `didUpdateEnhancedLocation` :sweat_smile:

File: Sources/AblyAssetTrackingInternal/AblyWrapper/DefaultAbly.swift [hunk 2]
```diff
@@ -356,6 +370,36 @@ extension DefaultAbly: AblyPublisher {
         }
     }
     
+    public func sendResolution(
+        trackable: Trackable,
+        resolution: Resolution,
+        completion: ResultHandler<Void>?
+    ) {
+        guard let channel = channels[trackable.id] else {
```
Comment (KacperKluka) [inline Sources/AblyAssetTrackingInternal/AblyWrapper/DefaultAbly.swift:378]: We should not send the resolution data via channel messages, we should be using channel's presence :sweat_smile: On Android, we also went with the channel messages approach, however, it generated a few issues and we've decided to switch to presence data (more details here https://github.com/ably/ably-asset-tracking-android/issues/638) :wink:

**Review by KacperKluka — CHANGES_REQUESTED** [review]
We should be sending the resolution data via channel presence data and not as channel messages. :wink: The requirements have changed and it looks like the iOS issue was not updated, sorry about that :disappointed:

File: Sources/AblyAssetTrackingPublisher/DefaultPublisherBuilder.swift [hunk 3]
```diff
@@ -22,7 +23,9 @@ class DefaultPublisherBuilder: PublisherBuilder {
                  routingProfile: RoutingProfile?,
                  delegate: PublisherDelegate?,
                  resolutionPolicyFactory: ResolutionPolicyFactory?,
-                 areRawLocationsEnabled: Bool?) {
+                 areRawLocationsEnabled: Bool?,
+                 isSendResolutionEnabled: Bool = false
```
Comment (KacperKluka) [inline Sources/AblyAssetTrackingPublisher/DefaultPublisherBuilder.swift:27]: Like above, I think that sending resolution should be enabled by default :wink:

File: Tests/SystemTests/ChannelModesTests.swift [hunk 4]
```diff
@@ -159,6 +159,8 @@ extension ChannelModesTests: SubscriberDelegate {
     
     func subscriber(sender: AblyAssetTrackingSubscriber.Subscriber, didChangeAssetConnectionStatus status: ConnectionState) {}
     
+    func subscriber(sender: AblyAssetTrackingSubscriber.Subscriber, didUpdateResolution resolution: Resolution) {}
+
```
Comment (KacperKluka) [inline Tests/SystemTests/ChannelModesTests.swift:163]: Since you made that method optional, can't we remove it? :wink:

**Review by KacperKluka — CHANGES_REQUESTED** [review]
Almost there :grin:

File: Sources/AblyAssetTrackingPublisher/DefaultPublisherBuilder.swift (hunk 3 above)
Comment (lukasz-szyszkowski) [reply Sources/AblyAssetTrackingPublisher/DefaultPublisherBuilder.swift]: changed a288514997ea39ae67ca98329302cc66012e4595

File: Sources/AblyAssetTrackingPublisher/DefaultPublisher.swift [hunk 5]
```diff
@@ -73,7 +74,8 @@ class DefaultPublisher: Publisher {
          routeProvider: RouteProvider,
          enhancedLocationState: TrackableState<EnhancedLocationUpdate> = TrackableState(),
          rawLocationState: TrackableState<RawLocationUpdate> = TrackableState(),
-         areRawLocationsEnabled: Bool? = nil
+         areRawLocationsEnabled: Bool? = nil,
+         isSendResolutionEnabled: Bool = false
```
[the remark this answers is not in the document]
Comment (lukasz-szyszkowski) [reply Sources/AblyAssetTrackingPublisher/DefaultPublisher.swift]: changed a288514997ea39ae67ca98329302cc66012e4595

File: Tests/SystemTests/ChannelModesTests.swift (hunk 4 above)
Comment (lukasz-szyszkowski) [reply Tests/SystemTests/ChannelModesTests.swift]: changed a288514997ea39ae67ca98329302cc66012e4595

**Review by KacperKluka — APPROVED** [review]
LGTM :+1:

File: Sources/AblyAssetTrackingSubscriber/DefaultSubscriber.swift [hunk 6]
```diff
@@ -297,4 +298,9 @@ extension DefaultSubscriber: AblySubscriberServiceDelegate {
         logger.debug("subscriberService.didReceiveEnhancedLocation.", source: String(describing: Self.self))
         callback(event: DelegateEnhancedLocationReceivedEvent(location: location))
     }
+    
+    func subscriberService(sender: AblySubscriber, didReceiveResolution resolution: Resolution) {
+        logger.debug("subscriberService.didReceiveResolution.", source: String(describing: Self.self))
```
Comment (maratal) [inline Sources/AblyAssetTrackingSubscriber/DefaultSubscriber.swift:303]: Cosmetics, but why not "\(Self.self)"?

Comment (lukasz-szyszkowski) [reply Sources/AblyAssetTrackingSubscriber/DefaultSubscriber.swift:303]: no reason :) if it's a blocker I can change it if not I would leave it as it is

**Review by maratal — APPROVED** [review]

**Pull request merged**